examples/pixels-admin/src/providers/authProvider.ts

```diff
@@ -1,128 +1,225 @@
-import { LegacyAuthProvider as AuthProvider } from "@pankod/refine-core";
+import { AuthBindings } from "@pankod/refine-core";
 
 import { supabaseClient } from "utility";
 
-export const authProvider: AuthProvider = {
+export const authProvider: AuthBindings = {
     login: async ({ email, password, providerName }) => {
-        // sign in with oauth
-        if (providerName) {
-            const { data, error } = await supabaseClient.auth.signInWithOAuth({
-                provider: providerName,
-            });
-
-            if (error) {
-                return Promise.reject(error);
+        try {
+            // sign in with oauth
+            if (providerName) {
+                const { data, error } =
+                    await supabaseClient.auth.signInWithOAuth({
+                        provider: providerName,
+                    });
+
+                if (error) {
+                    return {
+                        success: false,
+                        error,
+                    };
+                }
+
+                if (data?.url) {
+                    return {
+                        success: true,
+                    };
+                }
             }
 
-            if (data?.url) {
-                return Promise.resolve(false);
-            }
-        }
+            const { data, error } =
+                await supabaseClient.auth.signInWithPassword({
+                    email,
+                    password,
+                });
 
-        const { data, error } = await supabaseClient.auth.signInWithPassword({
-            email,
-            password,
-        });
+            if (error) {
+                return {
+                    success: false,
+                    error,
+                };
+            }
 
-        if (error) {
-            return Promise.reject(error);
-        }
+            const role = data.user?.app_metadata?.role;
 
-        const role = data.user?.app_metadata?.role;
+            if (role === "editor" || role === "admin") {
+                return {
+                    success: true,
+                };
+            }
 
-        if (role === "editor" || role === "admin") {
-            return Promise.resolve();
+            await supabaseClient.auth.signOut();
+
+            return {
+                success: false,
+                error: new Error("User not authorized."),
+            };
+        } catch (error) {
+            return {
+                success: false,
+                error: new Error("Login failed"),
```

Comment:
```suggestion
                error,
```

It might be better to throw an error from catch. but I do not know. catch can send an hard to read message.